Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -9,160 +9,37 @@
-{
-
-	// Prepare references to the docked ship's copy of the carrierShip's position and rotation for manipulation
-	Vector& carrierPos = mobiledockClients[dockedClientId].carrierPos;
-	Matrix& carrierRot = mobiledockClients[dockedClientId].carrierRot;
+unordered_map<wstring, CARRIERINFO> nameToCarrierInfoMap;
+unordered_map<wstring, DOCKEDCRAFTINFO> nameToDockedInfoMap;
+unordered_map<uint, CARRIERINFO*> idToCarrierInfoMap;
+unordered_map<uint, DOCKEDCRAFTINFO*> idToDockedInfoMap;
 
+uint saveFrequency = 7200; // once every 2 hours, possibly a heavy operation so doesn't have to happen often
+uint forgetCarrierDataInSeconds = 31556926; // since it's all kept in a single file, clear old enough values. Default value being a year
 
-	// If the carrier is out in space, simply set the undock location to where the carrier is currently
-	pub::SpaceObj::GetSystem(carrierShip, mobiledockClients[dockedClientId].carrierSystem);
-	pub::SpaceObj::GetLocation(carrierShip, carrierPos, carrierRot);
+// By default, docking is instant, but can be set to take defined number of seconds.
+uint dockingPeriod = 0;
 
-	carrierPos.x += carrierRot.data[0][1] * set_iMobileDockOffset;
-	carrierPos.y += carrierRot.data[1][1] * set_iMobileDockOffset;
-	carrierPos.z += carrierRot.data[2][1] * set_iMobileDockOffset;
-}
-
-// Overloaded function used with a specific carrier location, instead of extracting it from the ship itself
-void UpdateCarrierLocationInformation(uint dockedClientId, Vector pos, Matrix rot)
-{
-	// Prepare references to the docked ship's copy of the carrierShip's position and rotation for manipulation
-	Vector& carrierPos = mobiledockClients[dockedClientId].carrierPos;
-	Matrix& carrierRot = mobiledockClients[dockedClientId].carrierRot;
-
-	carrierPos = pos;
-	carrierRot = rot;
+// Above how much cargo capacity, should a ship be rejected as a docking user?
+int cargoCapacityLimit = 275;
 
-	carrierPos.x += carrierRot.data[0][1] * set_iMobileDockOffset;
-	carrierPos.y += carrierRot.data[1][1] * set_iMobileDockOffset;
-	carrierPos.z += carrierRot.data[2][1] * set_iMobileDockOffset;
-}
+float mobileDockingRange = 500.0f;
 
-inline void UndockShip(uint iClientID)
-{
-	// If the ship was docked to someone, erase it from docked ship list.
-	if (mobiledockClients[iClientID].mobileDocked)
-	{
-		uint carrierClientID = HkGetClientIdFromCharname(mobiledockClients[iClientID].wscDockedWithCharname);
+float maxDockingDistanceTolerance = 8.0f;
 
-		// If carrier is present at server - do it, if not - whatever. Plugin erases all associated client data after disconnect. 
-		if (carrierClientID != -1 && !mobiledockClients[iClientID].carrierDied)
-		{
-			wstring charname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
-			mobiledockClients[carrierClientID].mapDockedShips.erase(charname);
-			mobiledockClients[carrierClientID].iDockingModulesAvailable++;
-		}
-	}
+string defaultReturnSystem = "ew04"; //freeport 9, Omicron Theta

```

Comment:
Why is it set to this?

Reply:
It's a neutral-to-everyone-but-nomads base which is not particularly distant from any specific part of the sector thanks to connection to both Gamma and Alpha. In the actual dev build it'll be overriden via config to a different, even more faction-agnostic location.